pkg/operator/controller/ingress/load_balancer_service_test.go

```diff
@@ -57,5 +60,50 @@ func TestDesiredLoadBalancerService(t *testing.T) {
 		} else if !tc.expect && haveSvc {
 			t.Errorf("expected desiredLoadBalancerService to return nil service for endpoint publishing strategy type %v, got %#v", tc.strategyType, svc)
 		}
+
+		if tc.strategyType == operatorv1.LoadBalancerServiceStrategyType {
+			if infraConfig.Status.PlatformStatus != nil && infraConfig.Status.PlatformStatus.Type == configv1.AWSPlatformType {
+				checkServiceHasAnnotation(t, svc, awsLBProxyProtocolAnnotation, true, "*")
+				lb := &operatorv1.LoadBalancerStrategy{
+					ProviderParameters: &operatorv1.ProviderLoadBalancerParameters{
+						Type: operatorv1.AWSLoadBalancerProvider,
+						AWS: &operatorv1.AWSLoadBalancerParameters{
+							Type: operatorv1.AWSNetworkLoadBalancer,
+						},
+					},
+				}
+				ic.Status.EndpointPublishingStrategy.LoadBalancer = lb
```

Comment:
I think it would be better if the test case specified the `operatorv1.LoadBalancerStrategy` value as another input (along with the `strategyType` input).  The test case could also specify whether the annotation is expected.